cmd/cvp-trigger/main.go

```diff
@@ -383,3 +374,31 @@ func appendMultiStageParams(podSpec *v1.PodSpec, params map[string]string) {
 		podSpec.Containers[0].Args = append(podSpec.Containers[0].Args, fmt.Sprintf("--multi-stage-param=%s=%s", key, params[key]))
 	}
 }
+
+// getProwJobClient submits a new prow job and returns the client
+func getProwJobClient(prowjob *pjapi.ProwJob, config *prowconfig.Config) pjinterface.ProwJobInterface {
+	logrus.Info("getting cluster config")
+	clusterConfig, err := util.LoadClusterConfig()
+	if err != nil {
+		logrus.WithError(err).Fatal("failed to load cluster configuration")
+	}
+
+	logrus.WithFields(pjutil.ProwJobFields(prowjob)).Info("submitting a new prowjob")
+	var pjcset *pjclientset.Clientset
+	retryOptions := retry.Attempts(10)
+	err = retry.Do(
+		func() error {
+			pjcset, err = pjclientset.NewForConfig(clusterConfig)
```

Comment:
I don't think *this* is the thing that should be that robust? At least it's not what you describe in PR description, and it's not the issue I remember we were seeing. Your description says the problem is when the prowjob is submitted and *then* there's disruption (which can happen) betweeen `cvp-trigger` and the cluster, but prowjob runs fine. This means that what you need to make robust is handling the errors in setting up `Watch().

Reply:
You're right, I've updated the code to retry the `pjclient.Watch()` function instead